test/issue-679.cl

```diff
@@ -0,0 +1,40 @@
+// RUN: clspv %s -o %t2.spv
```

Comment:
I'd prefer to have a version of this that is more targeted at the pass. Maybe just run the lowering pass and the function internalizer

Reply:
I have added another test that target only the pass.